airflow/contrib/utils/parameters.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import absolute_import
+
+
+class XComParameter:
+    """
+    A parameter value that will be read from an XCom on execution of a task.
+
+    In the case of the kubernetes_operator, these will be converted into one or
+    more command line parameters in the script invocation. For AppEngine, these
+    will be sent... somehow to be determined later. Options include delimited
+    GET params, multiple GET params, or POST body.
+    """
+    def __init__(self, task_ids, xcom_key='return_value'):
+        """
+        Makes a new XComParameter
+
+        :param task_ids: a string representing task_id or a list of them
+        :param xcom_key: the key to read. defaults to 'return_value'
+        """
+        self.task_ids = task_ids
+        self.key = xcom_key
+
+    def get_values(self, ti):
+        """
+        Gets a list of parameter values
+
+        :param ti: A TaskInstance or other object implementing xcom_pull
+        :return: A container (list or tuple) with each returned value in the
+                 same order as the task_ids provided in the constructor. If a
+                 task_id has no value, it will be None.
+        """
+        if hasattr(self.task_ids, '__iter__') and not isinstance(self.task_ids, basestring):
+            return ti.xcom_pull(task_ids=self.task_ids, key=self.key)
+        else:
+            return [ti.xcom_pull(task_ids=self.task_ids, key=self.key)]
+
+
+def enumerate_parameters(source_dict, task_instance):
+    for key, value in source_dict.iteritems():
+        if isinstance(value, dict):
+            for inner_key, inner_value in value.iteritems():
+                yield (key, '{}={}'.format(inner_key, inner_value))
+        elif hasattr(value, '__iter__') and not isinstance(value, basestring):
```

Comment:
basestring objects don't have '__iter__'

Reply:
Strings have `__iter__` in Python 3. No reason to not future-proof here, especially since 2to3 and other compatibility linters will miss this.